spec/unit/actions/shared_examples_for_user_actions.rb

```diff
@@ -24,3 +24,20 @@
     subject.class.call(params)
   end
 end
+
+shared_examples 'when using unsuported sync_mode' do
+  before do
+    allow(project).to receive(:metadata).and_return(
+      'GOODOT_CUSTOM_PROJECT_ID' => 'project-123'
+    )
+    allow(project).to receive(:uri).and_return('project-uri')
+    allow(project).to receive(:add_data_permissions)
+    allow(domain).to receive(:clients).and_return([])
+    allow(data_source).to receive(:realize).and_return('filepath')
+    allow(File).to receive(:open).and_return("client_id\n123456789")
+  end
+
+  it 'fails' do
+    expect { subject.class.call(params) }.to raise_error
```

Comment:
Maybe we could check the exception message with a regex to make sure it is the one we expected?